pkg/curatedpackages/reader.go

```diff
@@ -0,0 +1,104 @@
+package curatedpackages
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	eksdv1 "github.com/aws/eks-distro-build-tooling/release/api/v1alpha1"
+	"sigs.k8s.io/yaml"
+
+	packagesv1 "github.com/aws/eks-anywhere-packages/api/v1alpha1"
+	"github.com/aws/eks-anywhere-packages/pkg/artifacts"
+	"github.com/aws/eks-anywhere/pkg/manifests"
+	releasev1 "github.com/aws/eks-anywhere/release/api/v1alpha1"
+)
+
+type PackageReader struct {
+	ManifestReader *manifests.Reader
+}
+
+func NewPackageReader(mr *manifests.Reader) *PackageReader {
+	return &PackageReader{
+		ManifestReader: mr,
+	}
+}
+
+func (r *PackageReader) ReadBundlesForVersion(version string) (*releasev1.Bundles, error) {
+	return r.ManifestReader.ReadBundlesForVersion(version)
+}
+
+func (r *PackageReader) ReadEKSD(eksaVersion, kubeVersion string) (*eksdv1.Release, error) {
+	return r.ManifestReader.ReadEKSD(eksaVersion, kubeVersion)
+}
+
+func (r *PackageReader) ReadImages(eksaVersion string) ([]releasev1.Image, error) {
+	images, err := r.ManifestReader.ReadImages(eksaVersion)
+	return images, err
+}
+
+func (r *PackageReader) ReadImagesFromBundles(b *releasev1.Bundles) ([]releasev1.Image, error) {
+	images, err := r.ManifestReader.ReadImagesFromBundles(b)
+	for _, v := range b.Spec.VersionsBundles {
+		images = append(images, v.PackageControllerImage()...)
+	}
+	return images, err
+}
+
+func (r *PackageReader) ReadCharts(eksaVersion string) ([]releasev1.Image, error) {
+	images, err := r.ManifestReader.ReadCharts(eksaVersion)
+	return images, err
+}
+
+func (r *PackageReader) ReadChartsFromBundles(ctx context.Context, b *releasev1.Bundles) []releasev1.Image {
+	images := r.ManifestReader.ReadChartsFromBundles(ctx, b)
+	for _, vb := range b.Spec.VersionsBundles {
+		artifact := GetPackageBundleRef(vb)
+		packages, err := FetchPackages(vb, ctx, artifact)
+		if err != nil {
+			fmt.Printf("error finding packages: %v", err)
+			continue
+		}
+		images = append(images, packages...)
+	}
+	return images
+}
+
+func FetchPackages(versionsBundle releasev1.VersionsBundle, ctx context.Context, art string) ([]releasev1.Image, error) {
+	data, err := Pull(ctx, art)
+	ctrl := versionsBundle.PackageController.Controller
+	if err != nil {
```

Comment:
might want to move this up before `ctrl := versionsBundle.PackageController.Controller`